net/bridge.go

```diff
@@ -51,3 +52,31 @@ func isDatapath(link netlink.Link) bool {
 		return false
 	}
 }
+
+func DetectHairpin(portIfName string, log *logrus.Logger) {
+	link, err := netlink.LinkByName(portIfName)
+	if err != nil {
+		log.Errorf("Unable to find link %q: %s", portIfName, err)
+	}
+
+	ch := make(chan netlink.LinkUpdate)
+	// See EnsureInterface for why done channel is not passed
+	if err := netlink.LinkSubscribe(ch, nil); err != nil {
+		log.Errorf("Unable to subscribe to netlink updates: %s", err)
+	}
+
+	pi, err := netlink.LinkGetProtinfo(link)
```

Comment:
it seems to me that this will crash if `link` is invalid; we should `return` if we hit the first error